Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -858,6 +939,59 @@ AXTextMarkerRange AXTextMarker::lineRange(LineRangeType type) const
     RELEASE_ASSERT_NOT_REACHED();
 }
 
+AXTextMarkerRange AXTextMarker::wordRange(WordRangeType type) const
+{
+    if (!isValid())
+        return { { }, { } };
+
+    AXTextMarker startMarker, endMarker;
+
+    if (type == WordRangeType::Right) {
+        startMarker = nextWordStart();
+        endMarker = startMarker.nextWordEnd();
+
+        if (startMarker == *this) {
+            auto resultRange = AXTextMarkerRange { startMarker, endMarker };
+            auto resultRangeString = resultRange.toString();
+            // A newline should not be returned as the right word (if we are at the end of a paragraph)
+            if (resultRangeString.length() && resultRangeString[0] == '\n')
+                return { WTFMove(*this), WTFMove(*this) };
```

Comment:
You can't move something twice, and moving `*this` inside its own function seems dangerous. Let's use copies instead:

`return  { *this, *this }`